Etg.Yams/DIModule.cs

```diff
@@ -60,7 +60,23 @@ public static void RegisterTypes(IUnityContainer container, YamsConfig config)
 
             RegisterDeploymentWatcher(container);
 
-            RegisterRemoteDirectory(container);
+			RegisterDeploymentRepository(container);
+        }
+
+        private static void RegisterDeploymentRepository(IUnityContainer container)
```

Comment:
Agrred. As you mentioned though, implementing the plugin architecture (i.e. providers) is probably higher priority.